### ISSUE TITLE
Custom padding for convolutions

### ISSUE THREAD

**Issue by jeandut** [tensorflow/tensorflow#1322]
For the moment it looks like TF only supports VALID or SAME convolutions ? Is there a clever work around to do custom padding convolutions like having a padding of length h defined by the user like in this [neon implementation](http://neon.nervanasys.com/docs/latest/generated/neon.layers.layer.Conv.html) ? Is implementing it in the pipeline ?

**Comment by Dringite** [comment]
Define custom. I can only imagine that you want to use something other than zeros in the padding, in which case you can use VALID and manually pad the feature maps.

**Comment by jeandut** [comment]
Actually I meant having a padding equals to 1 for instance, but you are right I will edit my question it is too vague.

**Comment by jeandut** [comment]
But as you suggested I could just add a column of 0 and use VALID convolutions, would the backprop still work though ?

**Comment by Dringite** [comment]
I'm confused as to why you would even need this. Doing a 3x3 kernel conv with SAME does padding of 1. 5x5 does padding of 2, etc...

**Comment by jeandut** [comment]
I wanted non symmetrical padding like only one side is padded but I guess this is not a common thing and neon is actually doing symmetrical, I am therefore closing this issue.

**Comment by abcdabcd987** [comment]
@jeandut how about <http://stackoverflow.com/questions/37659538/custom-padding-for-convolutions-in-tensorflow>?

**Comment by austingg** [comment]
just use tf.pad op for you purpose
